api/python/cell_census/src/cell_census/get_anndata.py

```diff
@@ -1,88 +1,27 @@
     {
-        "assay": NotRequired[Union[str, List[str]]],
-        "assay_ontology_term_id": NotRequired[Union[str, List[str]]],
-        "cell_type": NotRequired[Union[str, List[str]]],
-        "cell_type_ontology_term_id": NotRequired[Union[str, List[str]]],
-        "development_stage": NotRequired[Union[str, List[str]]],
-        "development_stage_ontology_term_id": NotRequired[Union[str, List[str]]],
-        "disease": NotRequired[Union[str, List[str]]],
-        "disease_ontology_term_id": NotRequired[Union[str, List[str]]],
-        "donor_id": NotRequired[Union[str, List[str]]],
-        "is_primary_data": NotRequired[bool],
-        "self_reported_ethnicity": NotRequired[Union[str, List[str]]],
-        "self_reported_ethnicity_ontology_term_id": NotRequired[Union[str, List[str]]],
-        "sex": NotRequired[Union[str, List[str]]],
-        "sex_ontology_term_id": NotRequired[Union[str, List[str]]],
-        "suspension_type": NotRequired[Union[str, List[str]]],
-        "tissue": NotRequired[Union[str, List[str]]],
-        "tissue_ontology_term_id": NotRequired[Union[str, List[str]]],
-        "tissue_general": NotRequired[Union[str, List[str]]],
-        "tissue_general_ontology_term_id": NotRequired[Union[str, List[str]]],
+        "obs": Optional[str],
+        "var": Optional[str],
     },
 )
 
-VarQuery = TypedDict(
-    "VarQuery",
-    {
-        "feature_id": NotRequired[Union[str, List[str]]],
-        "feature_name": NotRequired[Union[str, List[str]]],
-    },
-)
-
-
-def _build_query(
-    joinids: Optional[AxisCoordinate], query_defn: Optional[Union[ObsQuery, VarQuery]] = None
-) -> Optional[AxisQuery]:
-    """
-    Build a AxisQuery value filter from the user-defined query parameters.
-    """
-    if joinids is not None and query_defn is not None:
-        raise ValueError("Only one of joinids or value filter may be specified in a query.")
-
-    if joinids is not None:
-        return AxisQuery(coords=(joinids,))
-
-    if query_defn is None:
-        return None
-
-    query_conditions = []
-    for name, val in query_defn.items():
-        if isinstance(val, str):
-            query_conditions.append(f"{name} == '{val}'")
-        elif isinstance(val, numbers.Number):
-            query_conditions.append(f"{name} == {val}")
-        elif isinstance(val, list):
-            query_conditions.append(f"{name} in {val}")
-        else:
-            raise TypeError("Query must be string or list of strings")
-
-    if len(query_conditions) == 0:
-        return None
-
-    return AxisQuery(value_filter=" and ".join(query_conditions))
-
 
 def get_anndata(
     census: soma.Collection,
     organism: str,
     measurement_name: str = "RNA",
     X_name: str = "raw",
-    obs_joinids: Optional[AxisCoordinate] = None,
-    obs_value_filter: Optional[ObsQuery] = None,
-    var_joinids: Optional[AxisCoordinate] = None,
-    var_value_filter: Optional[VarQuery] = None,
+    value_filter: Optional[AxisValueFilters] = None,
```

Comment:
Any reason not to keep obs and var filters as separate params? It's how ExperimentQuery takes them, and would eliminate the intermediate `AxisValueFilters` type.  We could also have `AxisValueFilter` (singular) as an alias for `str`, if that has any semantic value.

We could then also use `AxisQuery()` (the "null" filter) for obs and var params, cleaning up some of the `... if ... else ...` lines below.

Reply:
We had them in a dict for `column_names`, so I replicated that pattern.  I don't mind breaking them out into multiple parameters, but we need to do the same for both `value_filter` and `column_names`.

I am personally agnostic so I will change (both) as you suggest.